cmd/oklog/forward_buffer.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"container/ring"
+	"sync"
+)
+
+type textScanner interface {
+	Scan() bool
+	Text() string
+	Err() error
+}
+
+// ringBuffer is a fixed-length ring buffer. It can be used by a forwarder, to 'drop' messages instead of applying backpressure. See Issue #15
+type ringBuffer struct {
+	//onward io.Writer
+	//prefix string
+	max int64
+
+	w *ring.Ring
+	r *ring.Ring
+
+	mutex     sync.Mutex
+	remaining int64
+	err       error
+}
+
+func (bf *ringBuffer) Scan() bool {
+	return true
+}
```

Comment:
As noted above, we're gonna want to propagate a shutdown signal thru somehow to terminate this.